### PR TITLE
Support connection / disconnection callback functions

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Add connection / disconnection callbacks params to login()
- Upgrade ain-js package to v1.12.0

Related issues:
- https://github.com/ainize-team/ainize-js/issues/125

### CODE REVIEW

File: src/handlers/handler.ts [hunk 1]
```diff
@@ -35,12 +36,15 @@ export default class Handler {
     console.log('Disconnected');
   }
 
-  private async disconnectedCb() {
+  private async connectionRetryCb(connectionCb?: ConnectionCallback, disconnectionCb?: DisconnectionCallback, webSocket?: any) {
     try {
+      if (disconnectionCb) {
+        disconnectionCb(webSocket);
```
Comment (jiyoung-an) [inline src/handlers/handler.ts:42]: where is the webSocket passed from?

Comment (platfowner) [reply src/handlers/handler.ts:42]: it's the last param of connectionRetryCb().

Comment (platfowner) [reply src/handlers/handler.ts:42]: https://github.com/ainblockchain/ain-js/blob/master/src/event-manager/event-channel-client.ts#L153

Comment (jiyoung-an) [reply src/handlers/handler.ts:42]: it seems that the webSocket object is not being passed in [this line](https://github.com/ainize-team/ainize-js/pull/126/files#diff-c015fc78f932d8b4de4f4a5568abbc291acf823c439ab951dd324089564a29cbR29), but that parameter appears to be required in the ain-js.

**Review by yoojinko — APPROVED** [review]
I tested it and it works well! Thanks!

**Comment by platfowner** [comment]
Thanks for the review! @yoojinko @jiyoung-an Now merging..

File: src/ainize.ts
Comment (jiyoung-an) [inline src/ainize.ts:1]: @platfowner shouldn't add a `disconnectioncb` when logging out?
https://github.com/ainize-team/ainize-js/pull/126/files#diff-18ba27879b90f0b7bcdc2205d931b9f01b9889c799fbdb63d430be0c013f7f01L61

Comment (jiyoung-an) [reply src/ainize.ts:1]: i've checked that the disconnection function in ain-js does not have a callback option.

Comment (platfowner) [reply src/ainize.ts:1]: (discussed offline) A disconnection callback can be added with login functions.

**Pull request merged**